src/ledger.ts

```diff
@@ -1280,3 +1310,237 @@ export class LedgerSignMessage extends LedgerBitcoinInteraction {
     });
   }
 }
+
+interface RegistrationConstructorArguments {
+  name: string;
+  braid: Braid;
+  policyHmac?: string;
+}
+
+/**
+ * A base class for any interactions that need to interact with a registered wallet
+ * by providing a base constructor that will generate the key origins and the policy
+ * from a given braid as well as methods for registering and returning a policy hmac
+ */
+export class LedgerBitcoinV2WithRegistrationInteraction extends LedgerBitcoinInteraction {
+  walletPolicy: MultisigWalletPolicy;
+
+  policyHmac?: Buffer;
+
+  policyId?: Buffer;
+
+  isLegacySupported = false;
+
+  isV2Supported = true;
```

Comment:
Maybe more definition is needed for these for completeness. `readonly`? `const`? `private` with a getter? It's best practice to keep attributes like this private while always providing getters/setters.

Reply:
Apparently this doesn't really work with the complex inheritance that this functionality is built on. I tried it and typescript complains.

Reply:
nvm, it works on the final descendent. So I'm making them readonly on `LedgerBitcoinV2WithRegistrationInteraction` and that settles it nicely.